akka-http-tests/src/test/scala/akka/http/scaladsl/OutgoingConnectionCancellationShouldKillEntityStreamSpec.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.{ Done, NotUsed }
+import akka.actor.ActorRef
+import akka.event.Logging
+import akka.http.scaladsl.client.RequestBuilding
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives
+import akka.http.scaladsl.settings.ConnectionPoolSettings
+import akka.stream._
+import akka.stream.scaladsl.{ Flow, Keep, Sink, Source }
+import akka.stream.stage.GraphStageLogic.StageActor
+import akka.stream.stage._
+import akka.testkit.{ AkkaSpec, ImplicitSender }
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future, Promise }
+import scala.util.{ Success, Try }
+
+/**
+ * Spec for https://github.com/akka/akka-http/issues/983
+ */
+class OutgoingConnectionCancellationShouldKillEntityStreamSpec extends AkkaSpec with ScalaFutures
```

Comment:
Hm indeed it seems `close the connection if response entity stream has been cancelled` is about this behavior, I'll dig deeper to see what the difference really is between the test now and that one.

I do like starting out with a high level reproducer and working back from that. Yeah, I can remove the this test once I have made a lower level one that shows off the issue properly.